crypto/src/signer.rs

```diff
@@ -37,16 +37,20 @@ impl EcdsaPub {
 pub type HmacSig = CryptoBuf;
 
 /// A common base struct that can be used for all digests, signatures, and keys.
+#[derive(Debug, PartialEq, Eq)]
 pub struct CryptoBuf(ArrayVec<u8, { Self::MAX_SIZE }>);
 
 impl CryptoBuf {
     pub const MAX_SIZE: usize = AlgLen::MAX_ALG_LEN_BYTES;
 
     pub fn new(bytes: &[u8], algs: AlgLen) -> Result<CryptoBuf, CryptoError> {
```

Comment:
It's a little odd that `algs` is passed into this function at all. If `bytes` is less than the max size of the CryptoBuf, I'm wondering if we should just create it. All off the callsites that use it need to check the size anyway. What do you think?

Reply:
SGTM. Done.